### PR TITLE
merge of AB test + FF + goal to develop with experiments

### PR DESCRIPTION
**Pull request by sufka861**
allready resolved merge conflicts togethere with Peer

### CODE REVIEW

File: Service/feature.logic.js
Comment (AtomicManic) [inline Service/feature.logic.js:13]: so the logic here is if the attributes match we show on, and otherwise we show off?

File: Service/utils.js
Comment (AtomicManic) [inline Service/utils.js:23]: add validation on req.clientIp

File: Service/utils.js
Comment (AtomicManic) [inline Service/utils.js:29]: validation on req.headers missing

File: repositories/experiment.repository.js
Comment (AtomicManic) [inline repositories/experiment.repository.js:28]: remove one !

File: repositories/experiment.repository.js
Comment (AtomicManic) [inline repositories/experiment.repository.js:31]: instead of returning null, you can do the check if(!experiment) throw new NotFoundEntity('experiment')

**Review by AtomicManic — APPROVED** [review]
Looking good. Made some comments to fix, but it's approved so after this you can merge

File: Service/feature.logic.js
Comment (sufka861) [reply Service/feature.logic.js:13]: if the attributes match - then show the feature "ON"
else
"OFF"

**Comment by sufka861** [comment]
all coments and bugs fixed

**Pull request merged**
